sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/InputFile.java

```diff
@@ -54,16 +55,18 @@
   }
 
   /**
-   * Path relative to module base directory. Path is unique and identifies file
-   * within given <code>{@link FileSystem}</code>. File separator is the forward
+   * Relative path to module (for normal Sensors) or project (for {@link SensorDescriptor#global() global} Sensors) base directory.
+   * File separator is the forward
    * slash ('/'), even on Microsoft Windows.
    * <br>
    * Returns <code>src/main/java/com/Foo.java</code> if module base dir is
    * <code>/path/to/module</code> and if file is
    * <code>/path/to/module/src/main/java/com/Foo.java</code>.
    * <br>
    * Relative path is not null and is normalized ('foo/../foo' is replaced by 'foo').
+   * @deprecated since 6.6 use {@link #inputStream()}, {@link #filename()} or {@link #uri()}
    */
+  @Deprecated
   @Override
   String relativePath();
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)